models/gtfs/ordem_servico_trips_shapes_gtfs2.sql

```diff
@@ -0,0 +1,448 @@
+          ) AS 'Domingo'
+        )
+      )
+  ),
+  -- 7. Despivota ordem de serviço por sentido
+  ordem_servico_sentido AS (
+    SELECT
+      *
+    FROM
+      ordem_servico_tipo_dia
+    UNPIVOT 
+    (
+      (
+        distancia_planejada,
+        partidas
+      ) FOR sentido IN (
+        (
+          extensao_ida,
+          partidas_ida
+        ) AS "I",
+        (
+          extensao_volta,
+          partidas_volta
+        ) AS "V"
+      )
+    )
+  ),
+  -- 8. Atualiza sentido dos serviços circulares na ordem de serviço
+  ordem_servico_sentido_atualizado AS (
+    SELECT 
+      o.* EXCEPT(sentido),
+      COALESCE(s.sentido, o.sentido) AS sentido
+    FROM 
+      ordem_servico_sentido AS o
+    LEFT JOIN
+      servico_trips_sentido AS s
+    USING
+      (feed_version, servico)
+    WHERE
+      distancia_planejada != 0
+      AND distancia_total_planejada != 0
+      AND partidas != 0
+  ),
+  -- 9. Busca anexo de trajetos alternativos
+  ordem_servico_trajeto_alternativo AS (
+    SELECT 
+      *
+    FROM 
+      {{ ref("ordem_servico_trajeto_alternativo_gtfs2") }}
+    {% if is_incremental() -%}
+      WHERE 
+        feed_start_date = "{{ var('data_versao_gtfs') }}"
+    {%- endif %}
+  ),
+  -- 10. Despivota anexo de trajetos alternativos
+  ordem_servico_trajeto_alternativo_sentido AS (
+    SELECT
+      *
+    FROM
+      ordem_servico_trajeto_alternativo
+    UNPIVOT 
+    (
+      (
+        distancia_planejada
+      ) FOR sentido IN (
+        (
+          extensao_ida
+        ) AS "I",
+        (
+          extensao_volta
+        ) AS "V"
+      )
+    )
+  ),
+  -- 11. Atualiza sentido dos serviços circulares no anexo de trajetos alternativos
+  ordem_servico_trajeto_alternativo_sentido_atualizado AS (
+  SELECT
+    * EXCEPT(sentido),
+    CASE
+      WHEN "C" IN UNNEST(sentido_array) THEN "C"
+      ELSE o.sentido
+    END AS sentido,
+  FROM
+    ordem_servico_trajeto_alternativo_sentido AS o
+  LEFT JOIN
+    (
+      SELECT
+        feed_start_date,
+        servico,
+        ARRAY_AGG(DISTINCT sentido) AS sentido_array,
+      FROM
+        ordem_servico_sentido_atualizado
+      GROUP BY
+        1,
+        2
+    ) AS s
+  USING
+    (feed_start_date, servico)
+  WHERE
+    distancia_planejada != 0
+  ),
+  -- 12. Trata a OS, inclui trip_ids e ajusta nomes das colunas
+  ordem_servico_tratada AS (
+  SELECT
+    *
+  FROM
+    (
+      (
+      SELECT
+        o.feed_version,
+        o.feed_start_date,
+        o.feed_end_date,
+        o.tipo_dia,
+        servico,
+        vista,
+        consorcio,
+        sentido,
+        distancia_planejada,
+        distancia_total_planejada,
+        inicio_periodo,
+        fim_periodo,
+        trip_id,
+        shape_id,
+      FROM
+        ordem_servico_sentido_atualizado AS o
+      LEFT JOIN
+        trips AS t
+      ON
+        t.feed_version = o.feed_version
+        AND o.servico = t.trip_short_name
+        AND 
+        CASE
```

Comment:
acredito que os cases no where podem ser trocados por OR.
exemplo:
(o.tipo_dia = t.tipo_dia OR o.tipo_dia = "Ponto Facultativo" AND t.tipo_dia = "Dia Útil" )